### PR TITLE
feat: Native HTTP port

### LINKED ISSUE #27: Native HTTP module port implementation
It would be nice to have a native Node.js HTTP module port implementation, as an example of a REST API without third-party frameworks / libs, in a functional way too.

I can help with this. =)

### PR DESCRIPTION
**Pull request by mxthevs**
closes #27

### CODE REVIEW

**Comment by mxthevs** [comment]
Hey @fdaciuk,
I think this is good to go, all API tests are passing now.
Maybe we can review this live? It's up to you. I do think that this implementation has space for improvements. 😄

**Comment by fdaciuk** [comment]
Great idea! Let's review this tomorrow morning at stream! 👯 👯 ✨

File: src/ports/native-http/modules/article.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    }
+  }),
+
+  'GET /api/articles/:slug': withTryAuth(async (request, response) => {
+    const payload = getPayload(request.auth)
+    const slug = 'slug'
+
+    pipe(
+      article.fetchArticle({
+        slug: request.params![slug] ?? '',
```
Comment (fdaciuk) [inline src/ports/native-http/modules/article.ts:51]: Use `?.` instead of `!`

File: src/ports/native-http/modules/article.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  }),
+
+  'PUT /api/articles/:slug': withAuth(async (request, response) => {
+    const payload = getPayload(request.auth)
+    const slug = 'slug'
+
+    for await (const body of request) {
+      const data: UpdateArticle = {
+        ...getPropFromRequestBody<UpdateArticle>(body, 'article'),
+        slug: request.params![slug] as Slug,
```
Comment (fdaciuk) [inline src/ports/native-http/modules/article.ts:79]: Use `?.` instead of `!`

File: src/ports/native-http/modules/article.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    }
+  }),
+
+  'DELETE /api/articles/:slug': withAuth(async (request, response) => {
+    const payload = getPayload(request.auth)
+    const slug = 'slug'
+
+    pipe(
+      article.deleteArticle({
+        slug: request.params![slug] ?? '',
```
Comment (fdaciuk) [inline src/ports/native-http/modules/article.ts:98]: Use `?.` instead of `!`

File: src/ports/native-http/modules/article.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  }),
+
+  'POST /api/articles/:slug/favorite': withAuth(async (request, response) => {
+    const payload = getPayload(request.auth)
+    const slug = 'slug'
+
+    pipe(
+      article.favoriteArticle({
+        userId: payload.id,
+        slug: request.params![slug] ?? '',
```
Comment (fdaciuk) [inline src/ports/native-http/modules/article.ts:126]: Use `?.` instead of `!`

File: src/ports/native-http/modules/article.ts [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  }),
+
+  'DELETE /api/articles/:slug/favorite': withAuth(async (request, response) => {
+    const payload = getPayload(request.auth)
+    const slug = 'slug'
+
+    pipe(
+      article.unfavoriteArticle({
+        userId: payload.id,
+        slug: request.params![slug] ?? '',
```
Comment (fdaciuk) [inline src/ports/native-http/modules/article.ts:140]: Use `?.` instead of `!`

File: src/ports/native-http/modules/profile.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import { withAuth, withTryAuth } from '../middlewares'
+
+const profileRoutes: Routes = {
+  'GET /api/profiles/:username': withTryAuth(async (request, response) => {
+    const username = 'username'
+    const payload = getPayload(request.auth)
+
+    pipe(
+      user.getProfile({
+        username: request.params![username] ?? '',
```
Comment (fdaciuk) [inline src/ports/native-http/modules/profile.ts:15]: Use `?.` instead of `!`

File: src/ports/native-http/modules/user.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+type GetUserFromRequestBody = (data: string) => CreateUser
+const getUserFromRequestBody: GetUserFromRequestBody = (data) => {
+  return pipe(
+    E.tryCatch(
+      () => JSON.parse(data),
+      E.toError,
+    ),
+    E.fold(
+      (error) => error,
+      (body) => body.user || {},
```
Comment (fdaciuk) [inline src/ports/native-http/modules/user.ts:19]: Use `??` instead of `||`

**Review by fdaciuk — CHANGES_REQUESTED** [review]
@mxthevs congrats man, great work!

**Review by fdaciuk — APPROVED** [review]

**Pull request merged**